rt/frontend/jaxrs/src/main/java/org/apache/cxf/jaxrs/impl/ResponseImpl.java

```diff
@@ -152,6 +152,9 @@ public boolean hasEntity() {
         Object actualEntity = getActualEntity();
         if (actualEntity == null) {
             return false;
+        } else if (entityBufferred) {
```

Comment:
When exception occurs while reading entity, it could happen that:
 -  getActualEntity() return `entity` (since `lastEntity` is `null`)
 - but the `entity` 's `InputStream` is fully consumed and is not empty
 - however, reading over fully consumed stream in this case is equivalent to reading over empty stream: nothing left to read

As such, the `hasEntity` return `false`. Adding the short circuit logic to make sure that if entity was buffered successfully, the response is considered to have an entity.